src/puppetlabs/puppetdb/scf/hash_debug.clj

```diff
@@ -35,18 +38,56 @@
   (map #(update-in % [:relationship] keyword) edges))
 
 (defn resource-params->kwds
-  "Updates the reousrce parameter keys from the DB to keywords for easier diffing
+  "Updates the resource parameter keys from the DB to keywords for easier diffing
    with the new catalog."
   [resources]
   (map (fn [resource]
          (update-in resource [:parameters] #(kitchensink/mapkeys keyword %)))
        resources))
 
+(defn query-resources
+  [certname
+   version]
+  (let [query-sql (resources/query->sql version ["=" "certname" certname])
+        results (resources/query-resources version query-sql "")]
+    (:result results)))
+
+(defn query-edges
+  [certname
+   version]
+  (let [query-sql (edges/query->sql version ["=" "certname" certname])
+        results (edges/query-edges version query-sql "")]
+    (:result results)))
+
+(defn expand-catalog-edges
+  [edges
+   certname
+   version]
+  (if (:data edges)
+    edges
+    (assoc edges :data (query-edges certname version))))
+
+(defn expand-catalog-resources
+  [resources
+   certname
+   version]
+  (if (:data resources)
```

Comment:
update-when-not perhaps. Umm, I've got two use-cases here only, can I flag this as a tiny issue and not bother for now or do we feel strongly about this?